src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BreakpointEvidence.java

```diff
@@ -0,0 +1,476 @@
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+/**
+ * Various types of read anomalies that provide evidence of genomic breakpoints.
+ */
+public class BreakpointEvidence implements Comparable<BreakpointEvidence> {
+    private final short contigIndex;
+    private final short eventWidth;
+    private final int contigStart;
+    private final String templateName;
+    private final TemplateEnd templateEnd;
+
+    public enum TemplateEnd {
+        UNPAIRED(""), PAIRED_UNKNOWN("/?"), PAIRED_FIRST("/1"), PAIRED_SECOND("/2"), PAIRED_INTERIOR("/0");
+
+        TemplateEnd( final String value ) { this.value = value; }
+
+        @Override
+        public String toString() { return value; }
+
+        private final String value;
+    }
+
+    /**
+     * evidence offset and width is set to "the rest of the fragment" not covered by this read
+     */
+    protected BreakpointEvidence( final GATKRead read, final ReadMetadata metadata ) {
+        final int templateLen = Math.round(metadata.getStatistics(read.getReadGroup()).getMedianFragmentSize());
+        final int width;
+        final int start;
+        if ( read.isReverseStrand() ) {
+            final int readStart = read.getStart();
+            width = readStart - (read.getUnclippedEnd() + 1 - templateLen);
+            start = readStart - width;
+        } else {
+            final int readEnd = read.getEnd() + 1;
+            width = read.getUnclippedStart() + templateLen - readEnd;
+            start = readEnd;
+        }
+        this.contigIndex = metadata.getContigID(read.getContig());
+        this.templateName = read.getName();
+        if ( templateName == null ) throw new GATKException("Read has no name.");
+        this.templateEnd = findTemplateEnd(read);
+        this.eventWidth = (short)width;
+        this.contigStart = start;
+    }
+
+    /**
+     * for use when the uncertainty in location has a fixed size
+     */
+    protected BreakpointEvidence( final GATKRead read, final ReadMetadata metadata,
+                        final int contigOffset, final short offsetUncertainty ) {
+        this.contigIndex = metadata.getContigID(read.getContig());
+        this.templateName = read.getName();
+        if ( templateName == null ) throw new GATKException("Read has no name.");
+        this.templateEnd = findTemplateEnd(read);
+        this.contigStart = contigOffset - offsetUncertainty;
+        this.eventWidth = (short)(2*offsetUncertainty);
+    }
+
+    /**
+     * for serialization
+     */
+    protected BreakpointEvidence( final Kryo kryo, final Input input ) {
+        this.contigIndex = input.readShort();
+        this.eventWidth = input.readShort();
+        this.contigStart = input.readInt();
+        this.templateName = kryo.readObject(input, String.class);
+        this.templateEnd = TemplateEnd.values()[input.readByte()];
+    }
+
+    /**
+     * to make a sentinel
```

Comment:
Will do.  But just to clear up the mystery, this constructor is called by the related serializer during deserialization.  I think this is a nice paradigm for getting around the Kryo issues:  every Kryo serializable specifies its own implementation of kryo.Serializer via an annotation (registration will happen automatically), and that implementation of kryo.Serializer is just a few lines of boilerplate that delegates to a serialize( Kryo, Output ) method in the object to serialize and to a constructor <init>( Kryo, Input ) to deserialize.  No magic reflection.  No end runs around access controls.  Just regular old calls to regular old Java methods.